src/components/ReanimatedDrawerLayout.tsx

```diff
@@ -0,0 +1,742 @@
+      edgeHitSlop,
+      fillHitSlop,
+      minSwipeDistance,
+      props.hideStatusBar,
+      props.keyboardDismissMode,
+      sideCorrectedDragX.value,
+      overlayDismissGesture,
+      drawerOpened,
+      isFromLeft,
+      containerWidth,
+      sideCorrection,
+      drawerState,
+      activeCursor,
+      enableTrackpadTwoFingerGesture,
+      mouseButton,
+      statusBarAnimation,
+    ]);
+
+    // When using RTL, row and row-reverse flex directions are flipped.
+    const reverseContentDirection = I18nManager.isRTL
+      ? isFromLeft
+      : !isFromLeft;
+
+    const dynamicDrawerStyles = {
+      backgroundColor: drawerBackgroundColor,
+      width: drawerWidth,
+    };
+
+    const containerStyles = useAnimatedStyle(() => {
+      if (drawerType === DrawerType.FRONT) {
+        return {};
+      }
+
+      return {
+        transform: [
+          {
+            translateX: interpolate(
+              openValue.value,
+              [0, 1],
+              [0, drawerWidth * sideCorrection],
+              Extrapolation.CLAMP
+            ),
+          },
+        ],
+      };
+    });
+
+    const drawerAnimatedStyle = useAnimatedStyle(() => {
+      const closedDrawerOffset = drawerWidth * -sideCorrection;
+
+      return {
+        transform: [
+          {
+            translateX:
+              drawerType === DrawerType.BACK
+                ? 0
+                : drawerState === DrawerState.IDLE
+                ? drawerOpened
+                  ? 0
+                  : closedDrawerOffset
+                : interpolate(
+                    openValue.value,
+                    [0, 1],
+                    [closedDrawerOffset, 0],
+                    Extrapolation.CLAMP
+                  ),
+          },
+        ],
+        flexDirection: reverseContentDirection ? 'row-reverse' : 'row',
+      };
+    });
+
+    const containerAnimatedProps = useAnimatedProps(() => ({
+      importantForAccessibility:
+        Platform.OS === 'android'
+          ? isDrawerOpen.value
+            ? ('no-hide-descendants' as const)
+            : ('yes' as const)
+          : undefined,
+    }));
+
+    const children =
+      typeof props.children === 'function'
+        ? props.children(openValue) // renderer function
+        : props.children;
+
+    const componentRef = React.useRef(null);
+
+    React.useImperativeHandle(
+      ref,
+      () => ({
+        ...componentRef,
+        openDrawer,
+        closeDrawer,
+      }),
+      // componentRef.current is neccessary, eslint thinks it will be reloaded each render
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      [componentRef.current, openDrawer, closeDrawer]
+    );
+
+    return (
+      <GestureDetector
+        gesture={panGesture}
+        userSelect={props.userSelect}
+        enableContextMenu={props.enableContextMenu}>
+        {/* ^^^ "false if MouseButton.RIGHT is specified" either remove that description or add that logic */}
```

Comment:
I think there's something to be done here :)

Reply:
This prop automatically switches to `false` when mouseButton is set to `MouseButton.RIGHT`, more details in [this PR](https://github.com/software-mansion/react-native-gesture-handler/pull/2676).

Tested on the `ReanimatedDrawerLayout` to confirm it works that way, couldn't test on the legacy one because it crashes on `web`.

Removed this comment in [7918726](https://github.com/software-mansion/react-native-gesture-handler/pull/3146/commits/7918726fa1a90f971704f6daf907b3cc28c19e11)

Reply:
>This prop automatically switches to false when mouseButton is set to MouseButton.RIGHT, more details in https://github.com/software-mansion/react-native-gesture-handler/pull/2676.

That's not exactly true. It behaves as it is `false`, but it's value is not set to `false`. You can see it [here](https://github.com/software-mansion/react-native-gesture-handler/blob/da9eed867ff09633c506fc9ad08634eaf707cbdb/src/web/tools/GestureHandlerWebDelegate.ts#L104).